### DISCUSSION TITLE
Build flag --type-check-only

### DISCUSSION THREAD

**Discussion by aovens-quantifi** [vercel/next.js#58266] [Ideas]
### Goals

1. Reduce CI times by only performing type check as opposed to full build

### Non-Goals

1. Implement a separate cli entrypoint

### Background

My NextJS project is getting quite large and the build is taking 2+ minutes to complete in CI. In my monorepo it is one project of many and therefore this time is quite significant. Ultimately I only care about the type check as the artifacts are thrown away (this is for a PR check only). When hacking this feature in locally, I observed < 30 second type check times.

### Proposal

Implement a new build flag --type-check-only (defaulted to false) which will bail the build as soon as type checking is finished

**Comment by aovens-quantifi** [comment]
Looking at the code, it probably wouldn't be too difficult to create a separate cli command instead which would avoid polluting the build command. This might be a better approach but curious of people's opinions

**Comment by 20manas** [comment]
+1. This is a good idea since it allows checking if the build will fail or not beforehand. Currently, next js linting is incomplete since the types which are generated during build time may not be present when `next lint` is run.
